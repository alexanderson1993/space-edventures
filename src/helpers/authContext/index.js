import React from "react";

const AuthContext = React.createContext({
  user: null,
  login: () => {},
  logout: () => {}
});

export default AuthContext;
