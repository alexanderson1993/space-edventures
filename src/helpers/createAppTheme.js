import { lighten, darken } from "polished";

const generateColor = color => ({
  base: color,
  light: lighten(0.2, color),
  dark: darken(0.2, color)
});

const generateBackground = color => ({
  level0: color,
  level1: lighten(0.015, color),
  level2: lighten(0.03, color),
  level3: lighten(0.045, color)
});

export default (theme = {}) => ({
  ...theme,
  color: {
    primary: generateColor(theme.colorPrimary || "#30fffe"),
    header: generateColor(theme.colorHeader || "#a1ecfb"),
    control: generateColor(theme.colorControl || "#acf9fb"),
    ...theme.color
  },
  background: {
    primary: generateBackground(theme.backgroundPrimary || "#031212"),
    ...theme.background
  }
});
