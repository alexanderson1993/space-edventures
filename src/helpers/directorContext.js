import React from "react";
export const DirectorContext = React.createContext({});
