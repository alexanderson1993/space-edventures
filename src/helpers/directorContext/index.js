import React, { useContext, useMemo } from "react";
import propTypes from "prop-types";
import AuthContext from "../authContext";
import { Query } from "react-apollo";
import graphQLHelper from "../graphQLHelper";
import CENTER_DIRECTOR from "./centerDirector.graphql";
import { Match } from "@reach/router";
export const DirectorContext = React.createContext({});

const Provider = ({ me = {}, children }) => {
  const value = useMemo(() => me, [me]);
  return (
    <DirectorContext.Provider value={value}>
      {children}
    </DirectorContext.Provider>
  );
};
export const DirectorProvider = ({ children }) => {
  const { user } = useContext(AuthContext);
  return (
    <Match path="/director/:centerId/*">
      {({ match }) => (
        <Query
          query={CENTER_DIRECTOR}
          variables={{
            centerId: match && match.centerId,
            id: user && user.id
          }}
          skip={!user || !user.id}
        >
          {graphQLHelper(({ me }) => (
            <Provider me={me} children={children} />
          ))}
        </Query>
      )}
    </Match>
  );
};

DirectorProvider.propTypes = {
  children: propTypes.node
};
