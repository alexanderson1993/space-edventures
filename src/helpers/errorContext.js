import React, { useReducer, useMemo } from "react";
export const ErrorContext = React.createContext();

const initialState = { error: false };
function reducer(state, action) {
  switch (action.type) {
    case "errored":
      return {
        error: true,
        errorData: action.errorData,
        errorInfo: action.errorInfo
      };
    case "cleared":
      return { error: false, errorData: null, errorInfo: null };
    default:
      throw new Error();
  }
}

export const ErrorProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  const value = useMemo(() => ({ state, dispatch }), [state]);
  return (
    <ErrorContext.Provider value={value}>{children}</ErrorContext.Provider>
  );
};
