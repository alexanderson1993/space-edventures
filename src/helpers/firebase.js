import { firebase } from "@firebase/app";
import "@firebase/auth";

var config = {
  apiKey: process.env.GATSBY_FIREBASE_API_KEY,
  databaseURL: process.env.GATSBY_FIREBASE_DATABASE_URL,
  storageBucket: process.env.GATSBY_FIREBASE_STORAGE_BUCKET,
  authDomain: process.env.GATSBY_FIREBASE_AUTH_DOMAIN,
  messagingSenderId: process.env.GATSBY_FIREBASE_MESSAGE_SENDER_ID,
  projectId: process.env.GATSBY_FIREBASE_PROJECT_ID
};
let authItem;
let baseAuthItem;
if (typeof window !== "undefined") {
  firebase.initializeApp(config);
  authItem = firebase.auth();
  baseAuthItem = firebase.auth;
}
export const auth = authItem;
export const baseAuth = baseAuthItem;
