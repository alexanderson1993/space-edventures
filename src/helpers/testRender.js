import React from "react";
import { render as rtlRender } from "react-testing-library";
import { ThemeProvider, createTheme } from "@arwes/arwes";
import { SoundsProvider, createSounds } from "@arwes/sounds";
import createAppTheme from "./createAppTheme";
import { AdminContext } from "../layout/arwesProvider";
import AuthContext from "./authContext/index";

const normalTheme = createTheme(
  createAppTheme({
    typography: {
      headerSizes: {
        h1: 44
      }
    }
  })
);

const ArwesProvider = ({ children }) => {
  return (
    <AuthContext.Provider
      value={{
        completeMagicLinkSignin: () => Promise.resolve()
      }}
    >
      <SoundsProvider sounds={createSounds({})}>
        <ThemeProvider theme={normalTheme}>
          <AdminContext.Provider value={{ isAdmin: false }}>
            {children}
          </AdminContext.Provider>
        </ThemeProvider>
      </SoundsProvider>
    </AuthContext.Provider>
  );
};

export function render(ui) {
  return rtlRender(<ArwesProvider>{ui}</ArwesProvider>);
}
