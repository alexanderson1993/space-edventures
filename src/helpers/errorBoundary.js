import React, { Component } from "react";
import { Button } from "../components";
import styled from "@emotion/styled";
import { ErrorContext } from "./errorContext";

const Container = styled("div")`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
export default class ErrorBoundary extends Component {
  static contextType = ErrorContext;
  componentDidCatch(error, errorInfo) {
    // Catch errors in any components below and re-render with error message
    this.context.dispatch({ type: "errored", errorData: error, errorInfo });
  }
  refresh() {
    if (typeof window !== "undefined") {
      localStorage.clear();
      window.location.reload();
    }
  }
  render() {
    if (this.context.state.error) {
      // Error path
      if (this.props.render) return this.props.render;
      return (
        <div className="error-boundary">
          <Container>
            <h2>Error in the Space EdVentures app.</h2>
            <Button onClick={this.refresh}>Refresh</Button>

            <h3>
              If you know what you are looking for, you can check out what went
              wrong.
            </h3>
            <details style={{ whiteSpace: "pre-wrap" }}>
              {this.context.state.errorData &&
                this.context.state.errorData.toString()}
              <br />
              {this.context.state.errorInfo.componentStack}
            </details>
          </Container>
        </div>
      );
    }
    // Normally, just render children
    return this.props.children;
  }
}
