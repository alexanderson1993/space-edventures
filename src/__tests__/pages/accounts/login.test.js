import React from "react";
import { render } from "../../../helpers/testRender";
import Login from "../../../pages/accounts/login";
import { ErrorProvider } from "../../../helpers/errorContext";

describe("login", () => {
  test("snapshot", () => {
    const { container } = render(
      <ErrorProvider>
        <Login />
      </ErrorProvider>
    );
    expect(container).toMatchSnapshot();
  });
  test("signup error", () => {});
  test("magic link error", () => {});
  test("create account switch", () => {});
});
