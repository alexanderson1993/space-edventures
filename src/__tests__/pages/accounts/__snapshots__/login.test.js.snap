// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login snapshot 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-7 {
  width: 360px;
  max-width: 80%;
}

.emotion-0 {
  color: #30fffe;
  background: transparent;
  border: transparent;
  padding: 5px 10px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 100%;
}

.emotion-5 {
  margin-top: 20px;
}

.emotion-2 {
  text-align: center;
}

.emotion-4 {
  margin-bottom: 10px;
}

<div>
  <div
    class="emotion-9 emotion-10"
  >
    <form
      class="emotion-7 emotion-8"
    >
      <h2>
        <span
          class="Sounds(Words)-root-0-1-1 Sounds(Words)-root-0-1-7 Sounds(Words)-hide-0-1-5"
        >
          <span
            class="Sounds(Words)-children-0-1-2"
          >
            Login
          </span>
        </span>
      </h2>
      <div>
        <label
          for="email"
        >
          Email: 
        </label>
        <div
          style="display: block;"
        >
          <div
            class="Sounds(Frame)-root-0-1-8 Sounds(Frame)-root-0-1-26 Sounds(Frame)-exited-0-1-25"
            id="email"
            value=""
          >
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-28 Sounds(Frame)-borderLeft-0-1-12"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-28 Sounds(Frame)-borderRight-0-1-13"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-28 Sounds(Frame)-borderTop-0-1-14"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-28 Sounds(Frame)-borderBottom-0-1-15"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-29 Sounds(Frame)-cornerLT-0-1-17 Sounds(Frame)-cornerLT-0-1-30"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-29 Sounds(Frame)-cornerLB-0-1-18 Sounds(Frame)-cornerLB-0-1-31"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-29 Sounds(Frame)-cornerRT-0-1-19 Sounds(Frame)-cornerRT-0-1-32"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-29 Sounds(Frame)-cornerRB-0-1-20 Sounds(Frame)-cornerRB-0-1-33"
            />
            <div
              class="Sounds(Frame)-box-0-1-9 Sounds(Frame)-box-0-1-27"
            >
              <div
                class="Sounds(Frame)-children-0-1-10"
              >
                <input
                  class="emotion-0"
                  classes="[object Object]"
                  id="email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <label>
          Password: 
        </label>
        <div
          style="display: block;"
        >
          <div
            class="Sounds(Frame)-root-0-1-8 Sounds(Frame)-root-0-1-34 Sounds(Frame)-exited-0-1-25"
            value=""
          >
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-36 Sounds(Frame)-borderLeft-0-1-12"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-36 Sounds(Frame)-borderRight-0-1-13"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-36 Sounds(Frame)-borderTop-0-1-14"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-36 Sounds(Frame)-borderBottom-0-1-15"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-37 Sounds(Frame)-cornerLT-0-1-17 Sounds(Frame)-cornerLT-0-1-38"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-37 Sounds(Frame)-cornerLB-0-1-18 Sounds(Frame)-cornerLB-0-1-39"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-37 Sounds(Frame)-cornerRT-0-1-19 Sounds(Frame)-cornerRT-0-1-40"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-37 Sounds(Frame)-cornerRB-0-1-20 Sounds(Frame)-cornerRB-0-1-41"
            />
            <div
              class="Sounds(Frame)-box-0-1-9 Sounds(Frame)-box-0-1-35"
            >
              <div
                class="Sounds(Frame)-children-0-1-10"
              >
                <input
                  class="emotion-0"
                  classes="[object Object]"
                  type="password"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-5 emotion-6"
      >
        <div
          class="Sounds(Button)-root-0-1-42"
          style="display: block; text-align: center;"
          type="submit"
        >
          <div
            class="Sounds(Frame)-root-0-1-8 Sounds(Frame)-root-0-1-45 Sounds(Frame)-hover-0-1-21 Sounds(Frame)-exited-0-1-25"
          >
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-47 Sounds(Frame)-borderLeft-0-1-12"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-47 Sounds(Frame)-borderRight-0-1-13"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-47 Sounds(Frame)-borderTop-0-1-14"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-47 Sounds(Frame)-borderBottom-0-1-15"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-48 Sounds(Frame)-cornerLT-0-1-17 Sounds(Frame)-cornerLT-0-1-49"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-48 Sounds(Frame)-cornerLB-0-1-18 Sounds(Frame)-cornerLB-0-1-50"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-48 Sounds(Frame)-cornerRT-0-1-19 Sounds(Frame)-cornerRT-0-1-51"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-48 Sounds(Frame)-cornerRB-0-1-20 Sounds(Frame)-cornerRB-0-1-52"
            />
            <div
              class="Sounds(Frame)-box-0-1-9 Sounds(Frame)-box-0-1-46"
            >
              <div
                class="Sounds(Frame)-children-0-1-10"
              >
                <div
                  class="Highlight-root-0-1-53"
                >
                  <div
                    class="Highlight-children-0-1-55"
                  >
                    <button
                      class="Sounds(Button)-button-0-1-43 Sounds(Button)-button-0-1-44"
                    >
                      <span
                        class="Sounds(Words)-root-0-1-1 Sounds(Words)-root-0-1-57 Sounds(Words)-hide-0-1-5"
                      >
                        <span
                          class="Sounds(Words)-children-0-1-2"
                        >
                          Login
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-2 emotion-3"
        >
          <small>
            ‒ OR ‒
          </small>
        </div>
        <div
          class="Sounds(Button)-root-0-1-42 emotion-4"
          style="display: block; text-align: center;"
          type="button"
        >
          <div
            class="Sounds(Frame)-root-0-1-8 Sounds(Frame)-root-0-1-59 Sounds(Frame)-hover-0-1-21 Sounds(Frame)-exited-0-1-25"
          >
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-61 Sounds(Frame)-borderLeft-0-1-12"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-61 Sounds(Frame)-borderRight-0-1-13"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-61 Sounds(Frame)-borderTop-0-1-14"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-61 Sounds(Frame)-borderBottom-0-1-15"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-62 Sounds(Frame)-cornerLT-0-1-17 Sounds(Frame)-cornerLT-0-1-63"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-62 Sounds(Frame)-cornerLB-0-1-18 Sounds(Frame)-cornerLB-0-1-64"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-62 Sounds(Frame)-cornerRT-0-1-19 Sounds(Frame)-cornerRT-0-1-65"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-62 Sounds(Frame)-cornerRB-0-1-20 Sounds(Frame)-cornerRB-0-1-66"
            />
            <div
              class="Sounds(Frame)-box-0-1-9 Sounds(Frame)-box-0-1-60"
            >
              <div
                class="Sounds(Frame)-children-0-1-10"
              >
                <div
                  class="Highlight-root-0-1-53"
                >
                  <div
                    class="Highlight-children-0-1-55"
                  >
                    <button
                      class="Sounds(Button)-button-0-1-43 Sounds(Button)-button-0-1-58"
                    >
                      <span
                        class="Sounds(Words)-root-0-1-1 Sounds(Words)-root-0-1-68 Sounds(Words)-hide-0-1-5"
                      >
                        <span
                          class="Sounds(Words)-children-0-1-2"
                        >
                          Create Account
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Sounds(Button)-root-0-1-42"
          style="display: block; text-align: center;"
          type="button"
        >
          <div
            class="Sounds(Frame)-root-0-1-8 Sounds(Frame)-root-0-1-70 Sounds(Frame)-hover-0-1-21 Sounds(Frame)-exited-0-1-25"
          >
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-72 Sounds(Frame)-borderLeft-0-1-12"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-72 Sounds(Frame)-borderRight-0-1-13"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-72 Sounds(Frame)-borderTop-0-1-14"
            />
            <div
              class="Sounds(Frame)-border-0-1-11 Sounds(Frame)-border-0-1-72 Sounds(Frame)-borderBottom-0-1-15"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-73 Sounds(Frame)-cornerLT-0-1-17 Sounds(Frame)-cornerLT-0-1-74"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-73 Sounds(Frame)-cornerLB-0-1-18 Sounds(Frame)-cornerLB-0-1-75"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-73 Sounds(Frame)-cornerRT-0-1-19 Sounds(Frame)-cornerRT-0-1-76"
            />
            <div
              class="Sounds(Frame)-corner-0-1-16 Sounds(Frame)-corner-0-1-73 Sounds(Frame)-cornerRB-0-1-20 Sounds(Frame)-cornerRB-0-1-77"
            />
            <div
              class="Sounds(Frame)-box-0-1-9 Sounds(Frame)-box-0-1-71"
            >
              <div
                class="Sounds(Frame)-children-0-1-10"
              >
                <div
                  class="Highlight-root-0-1-53"
                >
                  <div
                    class="Highlight-children-0-1-55"
                  >
                    <button
                      class="Sounds(Button)-button-0-1-43 Sounds(Button)-button-0-1-69"
                    >
                      <span
                        class="Sounds(Words)-root-0-1-1 Sounds(Words)-root-0-1-79 Sounds(Words)-hide-0-1-5"
                      >
                        <span
                          class="Sounds(Words)-children-0-1-2"
                        >
                          Send Magic Link
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
